### PR TITLE
Whitelisted Notebooks For TOPMed Demo

### PR DESCRIPTION
**Pull request by ansingh7115**
Companion firecloud-develop PRs: 
[dev](https://github.com/broadinstitute/firecloud-develop/pull/795)
[local](https://github.com/broadinstitute/firecloud-develop/pull/794)
[fiab](https://github.com/broadinstitute/firecloud-develop/pull/796)

This PR adds a new workspace tab for Notebooks. This tab is whitelisted - only those emails on the whitelist will be able to see the tab or access the Notebooks tab. (You will not be able to see the Notebooks UI unless you are on this list) The whitelist is extremely short - only those working on Leonardo and the few users who will be giving the TOPMed demo. 

To be clear - this UI is NOT final. It is a make-shift, placeholder UI that puts the Leonardo service into the hands of select users faster so that we can demo it, get feedback, etc. The UI is definitely flawed - that is okay because it will be ripped out in the next few months and replaced with a "real" notebooks UI for when the whitelist is lifted. 

The Notebooks tab is simply a table and a button. The table shows all the Spark clusters in that namespace. There's a Delete button next to each cluster row in the table. There's a Create button that opens up a modal where you can specify the cluster information. Only the name of the cluster is required. 

**Issues That Need to be Fixed** (would like to discuss in person)
- Bug removing labels from the Create modal 
- When signed in as a whitelisted user, you can't see the notebooks tab. However, you CAN go to the url for the Notebooks tab - it shows an Unauthorized error - but maybe it should just say that it doesn't recognize the url? 

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/workspace/notebooks/notebooks.cljs [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+   [broadfcui.endpoints :as endpoints]
+   [broadfcui.page.workspace.monitor.common :as moncommon]
+   [broadfcui.utils :as utils]
+   [broadfcui.components.foundation-dropdown :as dropdown]
+   [clojure.set]
+   [clojure.string :as string]
+   [dmohs.react :as react]))
+
+
+(def machineTypes ["n1-standard-1"
```
Comment (ansingh7115) [inline src/cljs/main/broadfcui/page/workspace/notebooks/notebooks.cljs]: This is hardcoded in for now - in the future we'd like to possibly have a leo endpoint that returns the machine type choices

File: src/cljs/main/broadfcui/common/input.cljs [hunk 2]
```diff
@@ -56,7 +56,7 @@
   {:test (comp not empty?) :message (str field-name " cannot be empty")})
 
 (defn integer [field-name & {:keys [min max]}]
-  (let [parses (partial re-matches #"\-?[0-9]+")
+  (let [parses (partial re-matches #"\-?[0-9]*")
```
Comment (ansingh7115) [inline src/cljs/main/broadfcui/common/input.cljs:59]: integer function is only used in notebooks - wanted to make it accept blanks

File: src/cljs/main/broadfcui/page/workspace/notebooks/notebooks.cljs [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+                        (icons/render-icon {} :delete)]))}
+
+                  {:header "Name" :initial-width 200
+                   :as-text :clusterName :sort-by :clusterName :sort-initial :asc
+                   :render
+                   (fn [cluster]
+                     (if (= (:status cluster) "Running")
+                       [:a {:style {:textDecoration "none"}
+                            :href (str (config/leonardo-url-root) "notebooks/" (:googleProject cluster) "/" (:clusterName cluster))
+                            :onClick #(utils/set-notebooks-access-token-cookie (utils/get-access-token))
```
Comment (ansingh7115) [inline src/cljs/main/broadfcui/page/workspace/notebooks/notebooks.cljs]: This sets the token in the new tab, the token will be auto-refreshed using a Notebooks extension

**Review by MattPutnam — APPROVED** [review]
👍

**Comment by ansingh7115** [comment]
jenkins retest

**Comment by ansingh7115** [comment]
jenkins retest

**Comment by ansingh7115** [comment]
jenkins retest

**Pull request closed without merging**